src/main/java/de/tum/in/www1/artemis/config/migration/setups/localvc/gitlab/MigrationEntryGitLabToLocalVC.java

```diff
@@ -0,0 +1,321 @@
+package de.tum.in.www1.artemis.config.migration.setups.localvc.gitlab;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+
+import org.eclipse.jgit.api.Git;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import de.tum.in.www1.artemis.config.migration.setups.localvc.LocalVCMigrationEntry;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.VcsRepositoryUri;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseStudentParticipation;
+import de.tum.in.www1.artemis.domain.participation.SolutionProgrammingExerciseParticipation;
+import de.tum.in.www1.artemis.domain.participation.TemplateProgrammingExerciseParticipation;
+import de.tum.in.www1.artemis.exception.localvc.LocalVCInternalException;
+import de.tum.in.www1.artemis.repository.AuxiliaryRepositoryRepository;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseRepository;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseStudentParticipationRepository;
+import de.tum.in.www1.artemis.repository.SolutionProgrammingExerciseParticipationRepository;
+import de.tum.in.www1.artemis.repository.TemplateProgrammingExerciseParticipationRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.UriService;
+import de.tum.in.www1.artemis.service.connectors.localvc.LocalVCRepositoryUri;
+import de.tum.in.www1.artemis.service.connectors.vcs.AbstractVersionControlService;
+
+@Component
+@Profile("gitlab")
```

Comment:
Should this component also only be active when the migration profile is active?

```suggestion
@Profile("gitlab & " PROFILE_MIGRATE_GITLAB_JENKINS_TO_LOCALVC)
```

Reply:
:heavy_check_mark: with 226fcdd9c8c46120325771ada92a09e35a8d7892